### PR TITLE
WIP: Initial prototype for adding MetalLB Operator

### PR DESCRIPTION
**Pull request by flaper87**
Signed-off-by: Flavio Percoco <flavio@redhat.com>

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by romfreiman** [comment]
not going to read the code here, but - what do u ask for user as an input?

**Comment by flaper87** [comment]
> not going to read the code here, but - what do u ask for user as an input?

It's all in here: https://github.com/openshift/assisted-service/pull/4708

File: internal/operators/metallb/manifest.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    workload.openshift.io/allowed: management
+spec: {}`
+
+const MetalLBOperatorGroup = `apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: "{{.OPERATOR_GROUP_NAME}}"
+  namespace: "{{.OPERATOR_NAMESPACE}}"`
+
+const MetalLBOperand = `apiVersion: metallb.io/v1beta1
```
Comment (ydayagi) [inline internal/operators/metallb/manifest.go:106]: @flaper87 according to https://metallb.universe.tf/configuration/#layer-2-configuration we need to define L2Advertisement as well:
`apiVersion: metallb.io/v1beta1
kind: L2Advertisement
metadata:
  name: cluster-api-ingress
  namespace: metallb-system
spec:
  ipAddressPools:
  - metallb-ingress-ip
  - metallb-api-vip`

File: internal/operators/metallb/manifest.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: metallb-api-vip
+  namespace: metallb
+spec:
+  protocol: layer2
+  autoAssign: false
+  addresses:
+    - $METALLB_IP/32
```
Comment (ydayagi) [inline internal/operators/metallb/manifest.go:121]: this should be METALLB_API_IP

File: internal/operators/metallb/manifest.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: metallb-ingress-ip
+  namespace: metallb
+spec:
+  protocol: layer2
+  autoAssign: false
+  addresses:
+    - $METALLB_IP/32
```
Comment (ydayagi) [inline internal/operators/metallb/manifest.go:132]: this should be METALLB_INGRESS_IP

File: internal/operators/metallb/manifest.go (hunk 2 above)
Comment (ydayagi) [inline internal/operators/metallb/manifest.go:121]: @flaper87 via [cloud.redhat.com](http://cloud.redhat.com/) the web interface asks for ingress and API IPs. It seems to me that we can use this input here.

**Pull request closed without merging**